activerecord/test/cases/associations/eager_load_nested_include_test.rb

```diff
@@ -99,6 +102,25 @@ def test_include_query
       end
     end
   end
+
+  def test_lazy_preload_query
+    res = ShapeExpression.all.merge!(lazy_preload: [ :shape, { paint: :non_poly } ])
+
+    assert_queries(1) { res.records }
+    assert_equal NUM_SHAPE_EXPRESSIONS, res.size
+
+    assert_queries(4) do
```

Comment:
That would be great to have some comments here: why four?

Also, shouldn't that be 2? One for `paint` records and one for `non_poly` records?

Reply:
Data for each class in this test is stored in separate tables, so there are a bit more queries than you have expected to see.

Reply:
Ah, ok.